searchEngine/SearchEngine.py

```diff
@@ -0,0 +1,55 @@
+from MockArticle import Article
+
+class SearchEngine:
+    """
+    This class implements the functions to score every article 
+    by the amount of incidences between keywords and synonyms to articles
+
+    Attributes:
+        keywords: A list of keywords
+        synonyms: A list of synonyms for the keywords
+        articles: A list of articles
+        keywords_weight: The weight that a keyword has on the scorig algorithm
+            Optional: default argument as 1
+        synonyms_weight: The weight that a synonym has on the scoring algorithm
+            Optional: default argument as 1
+    """
+    def __init__(self, articles):
+        """
+        Inits the search engine by loading all articles
+        """
+        self.articles = articles
+
+    def query(self, keywords, synonyms, keywords_weight=1, synonyms_weight=1):
```

Comment:
Do you foresee the same query being called multiple times often? If not, you might not need to store the keywords/synonyms here, and you can merge `query` and `score_articles` to be one function. 

That way your class interface involves a few less calls:

```python
    search = SearchEngine(articles)
    assert result == search.query(keywords_article_1, synonyms_article_1)
    assert result == search.query(keywords_article_2, synonyms_article_2)
```